src/Components/Components/src/CascadingParameterAttributeBase.cs

```diff
@@ -13,4 +13,12 @@ public abstract class CascadingParameterAttributeBase : Attribute
     /// of a cascading value.
     /// </summary>
     public abstract string? Name { get; set; }
+
+    /// <summary>
+    /// Gets a flag indicating whether the cascading parameter should
+    /// be supplied only once per component.
+    /// </summary>
+    private protected virtual bool SingleDelivery => false;
+
+    internal bool SingleDeliveryInternal => SingleDelivery;
```

Comment:
I think if you did internal, it would just work. https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/access-modifiers#summary-table

Sorry if I misled you with the `private protected` I did not realize that this needed to be accessed from outside the hiearchy.

Reply:
You're right. I don't know why I didn't just use `internal` in the first place. Changing this now.